python/caffe/pycaffe.py

```diff
@@ -109,8 +116,12 @@ def _Net_forward(self, blobs=None, start=None, end=None, **kwargs):
         outputs = set(self.outputs + blobs)
 
     if kwargs:
-        if set(kwargs.keys()) != set(self.inputs):
-            raise Exception('Input blob arguments do not match net inputs.')
+        expected_kwargs = set(kwargs.keys())
+        received_kwargs = set(self.inputs)
```

Comment:
Ah, I fixed this on one machine and forgot to commit it, sorry. Should be fixed now.